### ISSUE TITLE
Bibliography update not working when using build directory

### ISSUE THREAD

**Issue by Jamen1993** [texstudio-org/texstudio#856]
### Environment

- TeXstudio: 2.12.16 and 2.12.18
- Qt: 5.12.1
- OS: Windows 10 1909
- TeX distribution: texlive
- Language: German

### Expected behavior

TS is able to detect whether the bibliography was changed and decides if the bibliography tool must be run when doing a build.
I'd also expect this behaviour if I configured pdflatex and the bibliography tool to use a build directory.

### Actual behavior

The bibliography tool runs with every build, whether the bibliography was changed or not.

### How to reproduce

1. Setup pdflatex and biber to use a build directory
![grafik](https://user-images.githubusercontent.com/17482299/71734926-5adcca00-2e4d-11ea-8527-6563146189de.png)
2. Create the build directory in your working directory
3. Create a basic project with bibliography

`File: test.tex`
```latex
\documentclass[11pt,a4paper]{article}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage[british]{babel}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{graphicx}

\usepackage{biblatex}
\bibliography{literature}

\begin{document}
	Test
	
	\appendix
	\nocite{*}
	\printbibliography
\end{document}
```
`File: literature.bib`
```bibtex
@article{test,
	author = {Bach, Sandra},
	title = {A very good article},
	journaltitle = {Bad Journal},
	date = {2019-08}
}

@article{test2,
	author = {Bach, Sandra},
	title = {A very good article again},
	journaltitle = {Bad Journal},
	date = {2019-10}
}

@article{test3,
	author = {Bach, Sandra},
	title = {A very good article once again},
	journaltitle = {Bad Journal},
	date = {2019-12}
}
```

4. Build the project twice and watch biber run the second time even though the bibliography has not changed.

**Comment by sunderme** [comment]
duplicat eto #855

**Comment by Jamen1993** [comment]
Are you sure? This doesn't work in 2.12.16 as well.

**Comment by MeanSquaredError** [comment]
@Jamen1993 
It might be resolved after #855 was fixed. So can you try it with the latest development build of TXS from
https://bintray.com/sunderme/texstudio/texstudio-win#files
and see if the problem with biber is fixed?

**Comment by Jamen1993** [comment]
I tried it with the portable version of the latest build `texstudio-win-3.00.00-202001021845-git_6b3fcae.zip`. The behaviour persists.

I added the configuration file, just in case.

[texstudio.zip](https://github.com/texstudio-org/texstudio/files/4021701/texstudio.zip)

**Comment by MeanSquaredError** [comment]
OK, thank you for the configuration. Let me check the TXS behaviour it on my computer.

**Comment by MeanSquaredError** [comment]
@Jamen1993 
From your configuration I see that you are using unix-style (separator is / ) and relative paths your config file for the PDF and log paths. More specifically you use `./out` Can you please make two experiments:

1. Temporarily replace the additional paths for PDF and logs with absolute windows-style paths, that is the additional paths should be
`C:\absolute\path\to\out`
and then let me know if that fixes your problem.

2. Temporarily remove the `./` from the additional paths for PDF and log, that is these additional paths should be just
`out`
and then let me know if that fixes your problem.

**Comment by Jamen1993** [comment]
Sure.

Both don't solve the problem.

I assume the "problem" arises due to TXS not having the concept of a build directory (like texmaker). The additional search paths circumvent this by allowing the log and pdf viewers to find the output files if the user decides to use a build directory nonetheless.

Maybe this should be marked as enhancement and postponed.
A quick fix could be to apply either of the two search paths to TXS' routine that looks for the generated bibliography's up-to-dateness in the project root directory.

**Comment by MeanSquaredError** [comment]
> A quick fix could be to apply either of the two search paths to TXS' routine that looks for the generated bibliography's up-to-dateness in the project root directory.
That is what TXS is doing (and has been doing in the recent versions), but there is some problem with the latest builds. I managed to reproduce the problem on my computer and I am investigating it. Once I find the cause of the problem and a suitable fix I will provide a PR for it.

**Comment by MeanSquaredError** [comment]
@Jamen1993 
I am investigating this issue.

Can you please delete all the files in your `out` directory and provide the following information:
1. All the messages from the TXS messages panel after the first time you run the build process.
2. The log file form your output directory after the first build. It should be called out\test.log if you compiled test.tex
3. All the messages from the TXS messages panel after the second time you run the build process.
4. The log file form your output directory after the second build. It should be called out\test.log if you compiled test.tex

Please note that out\test.log will be overwritten after each build, so you will need to copy it under a different name/location after each build.

**Comment by MeanSquaredError** [comment]
@Jamen1993 
In your initial message you say that the bibliography file is called `literatur.bib` and in the sample test.tex you have
`\bibliography{literature}`

Note that the bibliography file lacks the ending `e`. If you are trying to include a missing bibliography file, then you will have the behavior that you described - it will try to run biber each time.

Can you please check and let me know if the file is called `literatur.bib` or `literature.bib` and if you are actually including the correct filename from your .tex file.

**Comment by Jamen1993** [comment]
That's a mistake. The file is called literature.bib so everything fits. I have corrected the mistake in the first post.

I performed the actions as requested. I used the latest nightly `texstudio-win-3.00.00-202001032141-git_fd4a0bd` in its portable distribution.

[results.zip](https://github.com/texstudio-org/texstudio/files/4022498/results.zip)

**Comment by MeanSquaredError** [comment]
Thank you for the logs. I managed the reproduce the problem reliably with the latest windows build of TXS on Windows 7.

Given that it only shows on Windows and on Linux it runs fine, I suspect a buggy behavior of the Qt filesystem functions (I saw it in the past but could not reproduce it reliably). It will take me a day or so to prepare a proper build environment on Windows and then investigate the issue properly.

I will provide a bugfix once I figure it out.
